Recruit-Training/Basic-Recruit-Training/person/Person.cpp

```diff
@@ -0,0 +1,56 @@
+#include "Person.h"
+#include "main.cpp"
+Person::Person()
+{
+    name_="";
+    age_= nullptr;
+}
+Person::Person(const std::string& name, int age)
+{
+    name_= name;
+    age_ = new int[1];
+    *age_ = age;   
+}
+Person::~Person()
+{
+    free(age_);
+}
+int Person::getAge() const
+{
+    return *age_;
+}
+const std::string& Person::getName() const
+{
+  return name_; 
+}
+void Person::setName(const std::string& newName)
+{
+  name_ = newName; 
+}
+void Person::setAge(int newAge)
+{
+  if (age_ == nullptr)
+  {
+      age_ = new int[1];
+  }
+
+  *age_ = newAge;
+}
+void Person::printInfo() const 
+{
+  printf("%s", name_.c_str());
+  printf("%d", *age_);
```

Comment:
If you didn't know, this is the C way of printing things out. In C++ it looks a bit like `std::cout << "whatever" << std::endl;`.  It both works but there are some differences that you can read about in here: https://stackoverflow.com/questions/2872543/printf-vs-cout-in-c.